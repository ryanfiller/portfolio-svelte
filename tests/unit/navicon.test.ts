import { describe, it, expect, vi } from 'vitest';
import { render, fireEvent } from '@testing-library/svelte'

import Navicon from '/src/components/navicon.svelte'

describe('<Navicon />', () => {
	let component

	let open = false
	const mockToggle = vi.fn(() => open = !open)
	function props(overrides: any = {}) {
		return ({
			boundElement: undefined,
			forSize: 'whatever',
			open: open,
			toggle: mockToggle,
			...overrides
		})
	}

	it('renders consistently', () => {
		component = render(Navicon, props())
		expect(component.container).toMatchSnapshot()
  })

  it('opens / closes', () => {
		component = render(Navicon, props({ open: open}))
		let button

		button = component.getByTitle('show off canvas navigation')
		
		fireEvent.click(button)
		expect(mockToggle).toHaveBeenCalled()

		component.rerender(props({ open: open }))
		button = component.getByTitle('hide off canvas navigation')

		fireEvent.click(button)
		expect(mockToggle).toHaveBeenCalled()

		component.rerender(props({ open: open }))
		button = component.getByTitle('show off canvas navigation')
  })
})