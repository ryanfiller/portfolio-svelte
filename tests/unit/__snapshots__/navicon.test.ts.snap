// Vitest Snapshot v1

exports[`<Navicon /> > renders consistently 1`] = `
<body>
  <div>
    <button
      class="navicon s-gtbQPPg4bHUW"
      data-for-size="whatever"
      title="show off canvas navigation"
    >
      <label
        class="s-gtbQPPg4bHUW"
        for="show-right-drawer"
      />
    </button>
    <!--&lt;Navicon&gt;-->
  </div>
</body>
`;
